### PR TITLE
Window Function Unit Tests

### PR DESCRIPTION
**Pull request by ngov17**
I created TestPgRegressWindow.java using yb_pg_window.{sql, out} based on the PostgreSQL regress test framework window.out file. 

All the queries that were not working have been commented out in yb_pg_window.out(And all reported as an issue). For most of the queries, I added ORDER BY to ensure the same output. (outlined in issue-4832)

EDIT: Issue-4832, which addresses the queries that have not been working in the test, has been resolved and the updated code in yb_pg_window.sql has no queries commented out and all the tests are passing as expected.

### CODE REVIEW

File: src/postgres/src/test/regress/expected/yb_pg_window.out [hunk 1]
```diff
@@ -932,10 +1114,13 @@ FROM tenk1 WHERE unique1 < 10 WINDOW w AS (order by four) ORDER by 1,2,3;
...
---SELECT first_value(unique1) over w,
---last_value(unique1) over w, unique1, four
---FROM tenk1 WHERE unique1 < 10
---WINDOW w AS (order by four range between current row and unbounded following)  ORDER by 1,2,3,4,5;
+SELECT first_value(unique1) over w,
+last_value(unique1) over w, unique1, four
+FROM tenk1 WHERE unique1 < 10
+WINDOW w AS (order by unique1 range between current row and unbounded following)  ORDER by 1,2,3,4,5;
+ERROR:  ORDER BY position 5 is not in select list
+LINE 4: ...en current row and unbounded following)  ORDER by 1,2,3,4,5;
```
Comment (kmuthukk) [inline src/postgres/src/test/regress/expected/yb_pg_window.out:1122]: Is this an intentional test case where you expect provide a non-existent column position "5"?

**Comment by kmuthukk** [comment]
@ngov17 -- is this sentence << All the queries that were not working have been commented out in yb_pg_window.out(And all reported as an issue).  >> in the original description still accurate, or can it be removed?

File: src/postgres/src/test/regress/expected/yb_pg_window.out (hunk 1 above)
Comment (ngov17) [reply src/postgres/src/test/regress/expected/yb_pg_window.out]: It isn't intentional, I have changed it and made sure the output matches what I have calculated in my latest commit.

File: src/postgres/src/test/regress/expected/yb_pg_window.out [hunk 2]
```diff
@@ -0,0 +1,3841 @@
...
+  FROM (VALUES (1,true), (2,true), (3,false), (4,false), (5,true)) v(i,b)
+  WINDOW w AS (ORDER BY i ROWS BETWEEN CURRENT ROW AND 1 FOLLOWING);
+ i | b | bool_and | bool_or
+---+---+----------+---------
+ 1 | t | t        | t
+ 2 | t | f        | t
+ 3 | f | f        | f
+ 4 | f | f        | t
+ 5 | t | t        | t
+(5 rows)
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_window.out:3841]: Please add back the newline from the original test (yes, there are two newlines).

File: src/postgres/src/test/regress/yb_pg_window_schedule [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# src/test/regress/window_test_schedule
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_window_schedule:1]: Please update to filename.

File: src/postgres/src/test/regress/sql/yb_pg_window.sql [hunk 4]
```diff
@@ -0,0 +1,1297 @@
...
+SELECT lead(ten) OVER (PARTITION BY four ORDER BY ten), ten, four FROM tenk1 WHERE unique2 < 10 ORDER by 1,2,3;
+
+SELECT lead(ten * 2, 1) OVER (PARTITION BY four ORDER BY ten), ten, four FROM tenk1 WHERE unique2 < 10 ORDER by 1,2,3;
+
+SELECT lead(ten * 2, 1, -1) OVER (PARTITION BY four ORDER BY ten), ten, four FROM tenk1 WHERE unique2 < 10 ORDER by 1,2,3;
+
+SELECT first_value(ten) OVER (PARTITION BY four ORDER BY ten), ten, four FROM tenk1 WHERE unique2 < 10 ORDER by 1,2,3;
+
+-- last_value returns the last row of the frame, which is CURRENT ROW in ORDER BY window.
+SELECT last_value(four) OVER (ORDER BY unique1), ten, four FROM tenk1 WHERE unique2 < 10 ORDER by 1,2,3;
```
[the remark this answers is not in the document]
Comment (ngov17) [reply src/postgres/src/test/regress/sql/yb_pg_window.sql:78]: See: https://github.com/yugabyte/yugabyte-db/issues/4832

**Pull request merged**